File: frontend/src/typings/api/get-problem-details-response.ts
/* tslint:disable */
import { ProblemFullyPopulatedDto } from './problem-fully-populated-dto';
import { SubmissionDto } from './submission-dto';

export interface GetProblemDetailsResponse {
  problem: ProblemFullyPopulatedDto;
  submissions: Array<SubmissionDto>;
}

File: backend/crates/api/src/models/http/responses/get_problem_details_response.rs
use crate::models::problem::ProblemFullyPopulatedDto;
use crate::models::submission::SubmissionDto;

#[derive(serde::Serialize, serde::Deserialize, rocket_okapi::JsonSchema)]
pub struct GetProblemDetailsResponse {
    pub problem: ProblemFullyPopulatedDto,
    pub submissions: Vec<SubmissionDto>,
}
